# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IndexedDatabase.h
# Commit: 7c9ee023c8cc
# Full Hash: 7c9ee023c8cc72aab0269cebeb7a2f02c5281cd1
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-04-01 21:26:59
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileInfo. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65550
# ==============================================================================

diff -r 7e9d56eb485e -r 7c9ee023c8cc dom/indexedDB/IndexedDatabase.h
--- a/dom/indexedDB/IndexedDatabase.h	Wed Apr 01 09:52:54 2020 +0000
+++ b/dom/indexedDB/IndexedDatabase.h	Wed Apr 01 09:53:01 2020 +0000
@@ -13,6 +13,7 @@
 #include "nsTArray.h"
 #include "InitializedOnce.h"
 #include "FileInfoFwd.h"
+#include "SafeRefPtr.h"
 
 namespace mozilla {
 namespace dom {
@@ -110,7 +111,7 @@
   StructuredCloneFileParent& operator=(StructuredCloneFileParent&&) = delete;
 
   // In IndexedDatabaseInlines.h
-  StructuredCloneFileParent(FileType aType, RefPtr<FileInfo> aFileInfo);
+  StructuredCloneFileParent(FileType aType, SafeRefPtr<FileInfo> aFileInfo);
 
   // In IndexedDatabaseInlines.h
   ~StructuredCloneFileParent();
@@ -123,10 +124,10 @@
   const indexedDB::FileInfo& FileInfo() const { return ***mContents; }
 
   // In IndexedDatabaseInlines.h
-  RefPtr<indexedDB::FileInfo> FileInfoPtr() const;
+  SafeRefPtr<indexedDB::FileInfo> FileInfoPtr() const;
 
  private:
-  InitializedOnce<const Maybe<RefPtr<indexedDB::FileInfo>>> mContents;
+  InitializedOnce<const Maybe<SafeRefPtr<indexedDB::FileInfo>>> mContents;
 };
 
 struct StructuredCloneReadInfoBase {